packages/netdog/netdog.spec

```diff
@@ -0,0 +1,96 @@
+%global _cross_first_party 1
+%undefine _debugsource_packages
+
+Name: %{_cross_os}netdog
+Version: 0.1.1
+Release: 0%{?dist}
+Summary: Bottlerocket network configuration helper
+License: Apache-2.0 OR MIT
+URL: https://github.com/bottlerocket-os/bottlerocket
+
+Source2: run-netdog.mount
+Source3: write-network-status.service
+Source4: netdog-tmpfiles.conf
+Source5: disable-udp-offload.service
+Source6: 00-resolved.conf
+Source7: generate-network-config.service
+
+BuildRequires: %{_cross_os}glibc-devel
+Requires: %{_cross_os}netdog
+%description
+%{summary}.
+
+%package -n %{_cross_os}netdog-systemd-networkd
+Summary: Bottlerocket network configuration helper
+Provides: %{_cross_os}netdog = 2:
+Requires: %{_cross_os}systemd-networkd
+Requires: %{_cross_os}systemd-resolved
+%description -n %{_cross_os}netdog-systemd-networkd
+%{summary}.
+
+%package -n %{_cross_os}netdog-wicked
+Summary: Bottlerocket network configuration helper
+Provides: %{_cross_os}netdog = 1:
+Requires: %{_cross_os}wicked
+%description -n %{_cross_os}netdog-wicked
+%{summary}.
+
+%prep
+%setup -T -c
+%cargo_prep
+
+%build
+mkdir bin
+
+echo "** Build Netdog for SystemD NetworkD"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features systemd-networkd \
+    --target-dir ${HOME}/.cache/networkd
+echo "** Build Netdog for Wicked"
+%cargo_build --manifest-path %{_builddir}/sources/Cargo.toml \
+    -p netdog \
+    --features wicked \
+    --target-dir ${HOME}/.cache/wicked
+    
+%install
+install -d %{buildroot}%{_cross_bindir}
+install -d %{buildroot}%{_cross_tmpfilesdir}
+install -d %{buildroot}%{_cross_unitdir}
+install -d %{buildroot}%{_cross_libdir}
+install -p -m 0644 %{S:4} %{buildroot}%{_cross_tmpfilesdir}/netdog.conf
+install -p -m 0644 %{S:2} %{S:3} %{S:7} %{buildroot}%{_cross_unitdir}
+install -d %{buildroot}%{_cross_libdir}/systemd/resolved.conf.d
+install -p -m 0644 %{S:6} %{buildroot}%{_cross_libdir}/systemd/resolved.conf.d
+%if %{with vmware_platform}
```

Comment:
I think we may want this file to be included in yet another package which we include in `vmware` variants. Right now this package still requires conditional compilation on vmware.

Reply:
LIke a third subpackage netdog-vmware which we then require in the vmware variants ontop of the normal

Reply:
After looking at the specific service, while it is networking related, I wouldn't claim that it's necessarily part of netdog.

I think what we can do for now is move this back into `os` and opt to include it in a variant-specific `vmware` package

Reply:
like adding a vmware-config subpackage in os?

Reply:
Although it bloats the number of rpms a tad, I think I like the idea of a `netdog-vmware-config` rpm and possibly moving the rest of the shared config into its own rpm and only basically swapping the resulting binary and `Requires` directives for the resulting rpms. That would allow others to "bring their own" netdog binary without having to remember to find and collect all the right configuration files needed. This is more preference and can be persuaded either way on the approach. I don't know if we really expect anyone to ever mess with and provide their own netdog but I think its more about setting a pattern for how we group config files that might be shared among mutually exclusive binaries. If I'm hijacking this thread we can move it away from here too.